index.html

```diff
@@ -767,18 +850,155 @@ <h2><dfn>CaptureController</dfn></h2>
+                    </td>
+                  </tr>
+                  <tr>
+                    <td><dfn>[[\Source]]</dfn></td>
+                    <td><code>null</code></td>
+                    <td>
+                      Tracks with which source the [=capture-session=] is associated.
+                      It is distinct from {{CaptureController/[[IsBound]]}}, in that
+                      {{CaptureController/[[IsBound]]}} is set even if
+                      {{MediaDevices/getDisplayMedia()}} rejects, whereas {{CaptureController/[[Source]]}}
+                      will only be set if {{MediaDevices/getDisplayMedia()}} succeeds.
+                    </td>
+                  </tr>
+                  <tr>
+                    <td><dfn>[[\DisplaySurfaceType]]</dfn></td>
+                    <td><code>null</code></td>
+                    <td>
+                      Once capture starts, this will be set to the type of the captured [=display surface=].
+                    </td>
+                  </tr>
+                  <tr>
+                    <td><dfn>[[\InitialFocusLost]]</dfn></td>
+                    <td><code>false</code></td>
+                    <td>
+                      Ensures that calls to {{CaptureController/setFocusBehavior()}} fail
+                      without throwing if the user manually focuses another tab or native application.
+                    </td>
+                  </tr>
+                  <tr>
+                    <td><dfn>[[\FocusWindowClosed]]</dfn></td>
+                    <td><code>false</code></td>
+                    <td>
+                      Tracks explicit closure of the window of opportunity, which happens when
+                      {{CaptureController/setFocusBehavior()}} is called, or [=task to close the window of opportunity to set
+                      focus behavior | the task to close the window=] executes.
+                      As applications can easily avoid such errors, {{CaptureController/setFocusBehavior()}} throws 
+                      when this slot is set to <code>true</code>.
+                    </td>
+                  </tr>
+                </tbody>
+              </table>
+            </dd>
+            <dt>
+              <dfn>setFocusBehavior</dfn>
+            </dt>
             <dd>
               <p>
-                Create one internal slot: <dfn data-dfn-for="CaptureController">[[\isBound]]</dfn>,
-                initialized to <code>false</code>.
+                Run the following steps:
               </p>
+              <ol>
+                <li>
+                  <p>
+                    Let <var>focusBehavior</var> be the method's first argument.
+                  </p>
+                </li>
+                <li>
+                  <p>
+                    If {{CaptureController/[[IsBound]]}} is <code>false</code>,
+                    [=exception/throw=] an "{{InvalidStateError}}" {{DOMException}}.
```

Comment:
Because our discussions seemed to suggest reduced scoped had consensus, with the following journey disallowed:
```js
const captureController = new CaptureController;
captureController.setFocusBehavior("no-focus-change");
navigator.mediaDevices({video: true, controller: captureController});
```
If we're all interested in allowing, then I am happy to introduce this as a follow-up.

Reply:
I am interested in calling the method ahead of time. I'm also wondering why the method isn't idempotent?

I'm curious, why did we reject a setter instead of a method? E.g.

```js
const captureController = new CaptureController;
captureController.focusBehavior = "no-focus-change";
navigator.mediaDevices({video: true, controller: captureController});
```
To clarify, I'm a fan of using the right JS surface where applicable. I was a fan of method when it was a simple `focus()` method, and I consider "set-methods" a bit of an anti-pattern really. But if both of you have good reasons for that that you can share, I'm willing to compromise.

Reply:
(Note I moved this comment down to make it more visible)

Reply:
> (Note I moved this comment down to make it more visible)

Where's my response? Could you move it down here too? Feel free to edit this message.

Reply:
A setter still allows us to throw exceptions, the only difference is we would have to add a `"default"` value. My interest here is that this seems like a better pattern that helps inform design questions (at least for me) like:

1. should we allow call be settable ahead of time? yes
2. should it be idempotent? yes
3. is there a third inferred option implied by not calling it? yes (today's UA focus behavior)

Reply:
* With a setter, we can both set ahead of time as well as keep it idempotent.
* With a setter, we don't have to specify the default value.
* With a setter, we don't allow reading the value, which could be confusing to some apps, since this is a hint they set, and does not report back what the UA will actually do.
* With a setter, side-effects and throwing are more naturally expected.

Reply:
Btw, these are points which were in my original message. I don't think it's good to move your arguments around, it makes it hard to see my counter-arguments.

Reply:
I've hunted down my original responses too. Here:

> I am interested in calling the method ahead of time.

Okay, then let's add that as a follow-up PR, which I'll produce and upload for review after the current PR gets merged.

> I'm also wondering why the method isn't idempotent?

Because I don't expect this to be called more than once by a well-structured app, given how it can only be called within a very narrow window of opportunity. But the follow-up will change things so that we can call arbitrarily many times before gDM is called, with same/different values, and up to one time after gDM is called.

> I'm curious, why did we reject a setter instead of a method?

It's more natural for a method to (1) have side-effect and (2) throw when used at an inappropriate time.

Reply:
> It's more natural for a method to (1) have side-effect and (2) throw when used at an inappropriate time.

I don't think so. `video.srcObject = stream` has lots of side-effects, can throw, and is susceptible to timing (active/inactive streams).

> With a setter, we don't allow reading the value, which could be confusing to some apps, since this is a hint they set, and does not report back what the UA will actually do.

Note that by "setter" I mean attribute setter steps, the way the term is used in [e.g. HTML](https://html.spec.whatwg.org/multipage/interaction.html#dom-hidden), i.e. not a method. Thus there's necessarily a getter as well.

I don't think an attribute would be confusing. `controller.focusBehavior` could default to `"default"`, and be setable up to focus finalization steps, after which attempting to set it could produce `InvalidStateError`. At that point the UA could even fix the getter to return either `"focus-captured-surface"` (if focus was decided) or `"no-focus-change"` (if not) so the app can learn whether it was successful (which it currently cannot, even if setFocusBehavior() succeeds). 

That way, if we ever decide we want an event to fire when this has happened, we have the getter we'd need to abide by the [§ 7.7. Use plain Events for state](https://w3ctag.github.io/design-principles/#state-and-subclassing) design principle.

Reply:
> Note that by "setter" I mean attribute setter steps, the way the term is used in [e.g. HTML](https://html.spec.whatwg.org/multipage/interaction.html#dom-hidden), i.e. not a method. Thus there's necessarily a getter as well.

I know.

I also don't like that the getter would have to return something, while the UA default behavior might be different for tabs/windows, and so the UA might not actually be able to clarify its default behavior by returning a value. That's one more reason why I prefer a method (set-only) rather than a property (set+get).